src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ByValueMarshalKindSupportDescriptor.cs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.Interop
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IMarshallingGenerator.SupportsByValueMarshalKind(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/> through <see cref="GetSupport(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/>
+    /// </summary>
+    public record ByValueMarshalKindSupportDescriptor(
+        ByValueMarshalKindSupport InSupport, string? InSupportDetails,
+        ByValueMarshalKindSupport OutSupport, string? OutSupportDetails,
+        ByValueMarshalKindSupport InOutSupport, string? InOutSupportDetails)
+    {
+        /// <summary>
+        /// A default <see cref="ByValueMarshalKindSupportDescriptor"/> for by value parameters. [In] is allowed, but unnecessary. Out is not allowed.
+        /// </summary>
+        public static readonly ByValueMarshalKindSupportDescriptor ValueTypeParameterDefault = new ByValueMarshalKindSupportDescriptor(
+            InSupport: ByValueMarshalKindSupport.Unnecessary, InSupportDetails: SR.InAttributeOnlyIsDefault,
+            OutSupport: ByValueMarshalKindSupport.NotSupported, OutSupportDetails: SR.OutAttributeNotSupportedOnByValueValueTypeParameters,
+            InOutSupport: ByValueMarshalKindSupport.NotSupported, InOutSupportDetails: SR.OutAttributeNotSupportedOnByValueValueTypeParameters);
+
+        /// <summary>
+        /// A default <see cref="ByValueMarshalKindSupportDescriptor"/> for by value parameters. [In] is allowed, but unnecessary. Out is not allowed.
+        /// </summary>
+        public static readonly ByValueMarshalKindSupportDescriptor ReferenceTypeParameterDefault = new ByValueMarshalKindSupportDescriptor(
+            InSupport: ByValueMarshalKindSupport.Unnecessary, InSupportDetails: SR.InAttributeOnlyIsDefault,
+            OutSupport: ByValueMarshalKindSupport.NotSupported, OutSupportDetails: SR.OutAttributeNotSupportedOnByValueReferenceTypeParameters,
+            InOutSupport: ByValueMarshalKindSupport.NotSupported, InOutSupportDetails: SR.OutAttributeNotSupportedOnByValueReferenceTypeParameters);
```

Comment:
These two cases don't differ in behavior and the wording of their details messages is almost identical. How would you feel about merging these two cases?

Reply:
Yeah, it's probably worth merging.